### DISCUSSION TITLE
Conversion from PNG to HEIC gives... PNG

### DISCUSSION THREAD

**Discussion by FrontierDK** [ImageMagick/ImageMagick#6172] [General]
Hi there.

A simple question: why is it that if one tries to convert a .PNG file into a heic or heif file, you send up...with a png? The file type (name only) IS .heic or heif - but the actual file contents is a PNG.

It doesn't matter if one uses convert.exe or magick.exe

**Comment by snibgo** [comment]
What versions of convert or magick? I suspect they are old, or have not been built with the required library to recognise the `.HEIC` extension etc.

**Reply by FrontierDK** [reply]
Hi there, I am using the latest version available today.

_Version: ImageMagick 7.1.1-3 Q16-HDRI x64 c5d5e71:20230311 https://imagemagick.org
Copyright: (C) 1999 ImageMagick Studio LLC
License: https://imagemagick.org/script/license.php
Features: Cipher DPC HDRI OpenCL
Delegates (built-in): bzlib cairo freetype gslib heic jng jp2 jpeg jxl lcms lqr lzma openexr pangocairo png ps raqm raw rsvg tiff webp xml zip zlib
Compiler: Visual Studio 2022 (193431942)_

It is very easy to test the problem:
magick.exe test.png test.heic
or
magick.exe test.png test.heif

Heck, even this produces the exact same image data as the heic/heif
magick.exe test.png test2.png

**Reply by dlemstra** [reply]
There is no support for writing `heic` images on Windows due to licensing issues.

**Reply by FrontierDK** [reply]
> There is no support for writing `heic` images on Windows due to licensing issues.

Hi there, thank you for mentioning the exact reason.

**Reply by averyfreeman** [reply]
Might the `.HEIC` extension help with your issue? It's included in this bundle: https://apps.microsoft.com/detail/9pmmsr1cgpwg - hopefully that would allow it.

There's also Crushee, it's in the chocolatey repo, source: https://github.com/xanderfrangos/crushee
